exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -280,17 +280,17 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		return err
 	}
 
-	// Initialize the patch helper.
-	patchHelper, err := patch.NewHelper(clusterResourceSetBinding, r.Client)
-	if err != nil {
-		return err
-	}
+	patch := client.MergeFromWithOptions(clusterResourceSetBinding.DeepCopy(), client.MergeFromWithOptimisticLock{})
+
+	var errList []error
 
 	defer func() {
 		// Always attempt to Patch the ClusterResourceSetBinding object after each reconciliation.
-		if err := patchHelper.Patch(ctx, clusterResourceSetBinding); err != nil {
-			log.Error(err, "Failed to patch config")
+		if patchErr := r.Client.Patch(ctx, clusterResourceSetBinding, patch); patchErr != nil {
+			log.Error(patchErr, "Failed to patch clusterResourceSetBinding", "clusterResourceSetBinding", klog.KObj(clusterResourceSetBinding))
+			errList = append(errList, patchErr)
 		}
+		err = kerrors.NewAggregate(errList)
```

Comment:
Let's use the pattern we have elsewhere :-)
```suggestion
		if err := r.Client.Patch(ctx, clusterResourceSetBinding, patch); err != nil {
			rerr = kerrors.NewAggregate([]error{rerr, errors.Wrapf(err, "failed to patch ClusterResourceSetBinding %s", klog.KObj(clusterResourceSetBinding))})
		}
```

There's no need to log here and return an unwrapped error. I think it gets more clear to just return a wrapped error and get a single log line.